book/src/dev/rfcs/0004-asynchronous-script-verification.md

```diff
@@ -82,18 +91,157 @@ parallelism.
 # Reference-level explanation
 [reference-level-explanation]: #reference-level-explanation
 
-We add a `Request::AwaitUtxo(OutPoint)` and
-`Response::Utxo(transparent::Output)` to the state protocol. As described
-above, the request name is intended to indicate the request's behavior: the
-request does not resolve until the state layer learns of a UTXO described by
-the request.
+## Data structures
+[data-structures]: #data-structures
```

Comment:
Question: I'm not sure I understand why these references are created. Isn't writing `[something][section]` equivalent to writing `[something](#section)`? Or are there any checks that help with typos?

Reply:
This is the standard format used by Rust documentation:
- `[data-structures]:` creates an invisible link that we can reference using `[data-structures]` (no colon).
- `#data-structures` creates an invisible anchor that we can reference using `(#data-structures)`.

So `[something][section]` creates a visible link with text `something` to `[section]:`.

See https://github.github.com/gfm/#link-reference-definitions , specifically:
> The link destination may not be omitted

Reply:
Right, but it means that the `: #section` destination should not be omitted if a `[section]` reference is created. But you don't need that, you can just write `See also [Section](#section)`, which avoids having to create the redundant `[section]: #section`.

I'm just wondering if there's any advantage, because it seems like it could be error prone and have more verbose titles.

I might be missing something, like if the `#section` anchor isn't created without the `[section]: #section` or if `[section]` can be automatically checked for typos.

Reply:
I'd prefer to be flexible and allow links using both `[section]` and `[section](#section)`.

I don't think typos in anchors have a large enough impact to make automated checks worthwhile.
(But we should check again once Zebra is closer to audits and sending funds.)